tests/sweep_framework/runner.py

```diff
@@ -0,0 +1,334 @@
+                status, message = False, str(e)
+                e2e_perf = None
+            output_queue.put([status, message, e2e_perf])
+    except Empty as e:
+        ttnn.close_device(device)
+        exit(0)
+
+
+def get_timeout(test_module):
+    try:
+        timeout = test_module.TIMEOUT
+    except:
+        timeout = 30
+    return timeout
+
+
+def execute_suite(test_module, test_vectors, pbar_manager, suite_name):
+    results = []
+    input_queue = Queue()
+    output_queue = Queue()
+    p = None
+    timeout = get_timeout(test_module)
+    suite_pbar = pbar_manager.counter(total=len(test_vectors), desc=f"Suite: {suite_name}", leave=False)
+    for test_vector in test_vectors:
+        if DRY_RUN:
+            print(f"Would have executed test for vector {test_vector}")
+            continue
+        result = dict()
+        if deserialize(test_vector["validity"]) == VectorValidity.INVALID:
+            result["status"] = TestStatus.NOT_RUN
+            result["exception"] = "INVALID VECTOR: " + test_vector["invalid_reason"]
+            result["e2e_perf"] = None
+            result["timestamp"] = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+            results.append(result)
+            suite_pbar.update()
+            continue
+        else:
+            test_vector.pop("invalid_reason")
+            test_vector.pop("status")
+            test_vector.pop("validity")
+        if p is None:
+            p = Process(target=run, args=(test_module, input_queue, output_queue))
+            p.start()
+        try:
+            if MEASURE_PERF:
+                # Run one time before capturing result to deal with compile-time slowdown of perf measurement
+                input_queue.put(test_vector)
+                output_queue.get(block=True, timeout=timeout)
+            input_queue.put(test_vector)
+            response = output_queue.get(block=True, timeout=timeout)
+            status, message, e2e_perf = response[0], response[1], response[2]
+            if status:
+                result["status"] = TestStatus.PASS
+                result["message"] = message
+            else:
+                if "Out of Memory: Not enough space to allocate" in message:
+                    result["status"] = TestStatus.FAIL_L1_OUT_OF_MEM
+                elif "Watcher" in message:
+                    result["status"] = TestStatus.FAIL_WATCHER
+                else:
+                    result["status"] = TestStatus.FAIL_ASSERT_EXCEPTION
+                result["exception"] = message
+            if e2e_perf and MEASURE_PERF:
+                result["e2e_perf"] = e2e_perf
+            else:
+                result["e2e_perf"] = None
+        except Empty as e:
+            print(f"SWEEPS: TEST TIMED OUT, Killing child process {p.pid} and running tt-smi...")
+            p.terminate()
+            p = None
+            smi_dir = architecture.tt_smi_path(ARCH)
+            smi_process = subprocess.run([smi_dir, "-tr", "0"])
```

Comment:
we may want "all", "wait" if we're also resetting a multichip device. Can't seem to find the difference between "-tr" and "-wr"? Any reason why you chose "-tr"? Is that what it is for grayskull? For my wormhole I use "-wr"

Reply:
-tr was what I found in the tt-smi help. Let me know if you think it should be something different? Is there a good catch-all we should use or should it be configurable somehow maybe?
<img width="1012" alt="Screenshot 2024-07-24 at 8 42 25 PM" src="https://github.com/user-attachments/assets/29a9bb4d-609d-4101-87bd-e30a4a6ac78b">

Reply:
I use `-tr` for GS and `-wr` for WH, though I'm not sure what the specific differences are.

Reply:
I see, GS tt-smi is much older and doesn't have -wr option. I'll change the options to be arch specific. I'll go with "-wr" for WH, and "-tr" for GS.